### ISSUE TITLE
uv python find --system finds managed interpreters

### ISSUE THREAD

**Issue by bluss** [astral-sh/uv#8533]
Try:

```
uv python find --system
~/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/bin/python3
```

Help for `--system`: *Only find system Python interpreters.*

This is surprising behaviour. From other parts of Uv we learn that there are uv-managed and system interpreters, then we expect to get a system interpreter here. I.e /usr/bin/python or other python not installed by Uv.

```
      --python-preference <PYTHON_PREFERENCE>
          Whether to prefer uv-managed or system Python installations [env: UV_PYTHON_PREFERENCE=]
          [possible values: only-managed, managed, system, only-system]
```

@zanieb has explained how the existing behaviour is by design, if I understand correctly, but the naming of it all seems easy to misunderstand.

**Comment by zsol** [comment]
That's right, `--system` makes uv look for Python versions on the search path (i.e. `$PATH`). If there happens to be a uv-installed Python in there, that will be returned. If you want uv to ignore managed Python installations regardless of where they are, `--no-managed-python` is what gets you that behavior.

I do agree there's room for improvement in the help text though
